src/nvim/testdir/test_expr_utf8.vim

```diff
@@ -0,0 +1,38 @@
+" Tests for expressions using utf-8.
+if !has('multi_byte')
+  finish
+endif
+set encoding=utf-8
+scriptencoding utf-8
+
+func Test_strgetchar()
+  call assert_equal(char2nr('á'), strgetchar('áxb', 0))
+  call assert_equal(char2nr('x'), strgetchar('áxb', 1))
+
+  call assert_equal(char2nr('a'), strgetchar('àxb', 0))
+  call assert_equal(char2nr(''), strgetchar('àxb', 1))
```

Comment:
This should be testing that the composing character is returned separately.  You're missing the composing char in the `char2nr` call.

How are you merging the changes?